### DISCUSSION TITLE
Get query data of another component

### DISCUSSION THREAD

**Discussion by Gabriel-Passoss** [TanStack/query#8237] [Q&A]
I have a date-picker component that makes a request to my server based on the selected date. I would like to access this information that is cached in another component to build the statistics screen, but apparently the statistics component loads before finishing loading the sales information. Would it be possible for me to force the query key data to exist before displaying the component?

And I would like to avoid using 'use client' in their parent component, as Next does not recommend that an entire page be a client component.

![image](https://github.com/user-attachments/assets/b0757a6c-1117-4fe1-a22f-5a2b18636554)
![image](https://github.com/user-attachments/assets/b30433b4-d2fc-4cb2-9ca8-0a50512b8257)

**Comment by TkDodo** [comment]
`queryClient.getQueryData` is not reactive. What you want is to just call `useQuery` in every component that needs data: https://tkdodo.eu/blog/react-query-as-a-state-manager
